pasystem/pasystem-impl/impl/src/java/org/sakaiproject/pasystem/impl/rest/PASystemEntityProvider.java

```diff
@@ -183,6 +183,11 @@ public String getTimezoneToolUrlForUser() {
             User thisUser = UserDirectoryService.getCurrentUser();
             String userid = thisUser.getId();
 
+            // If there is no user (e.g. on the gateway site!) there's no timezone
+            if(userid == null || userid == "") {
```

Comment:
The issue is for the case where the browser isn't logged in at all. I don't know what the contract of getCurrentUser is, but null or "" seemed like the only likely choices so I checked for both.


On August 6, 2016 6:06:12 AM EDT, Steve Swinsburg <notifications@github.com> wrote:
>> @@ -183,6 +183,11 @@ public String getTimezoneToolUrlForUser() {
>>              User thisUser = UserDirectoryService.getCurrentUser();
>>              String userid = thisUser.getId();
>>  
>> +            // If there is no user (e.g. on the gateway site!)
>there's no timezone
>> +            if(userid == null || userid == "") {
>
>Could this please use StringUtils.isBlank/isEmpty?
>Also, why would there be a null/blank id returned? Shouldn't there just
>be no user?
>
>---
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/sakaiproject/sakai/pull/3142/files/4456f366aff0be548e4042e8ccbf8626091673fc#r73784008

Mitch Golden
CTO, Noodle Partners